apps/antalmanac/src/components/Header/SettingsMenu.tsx

```diff
@@ -1,96 +1,142 @@
+                >
+                    Light
+                </Button>
+                <Button
+                    startIcon={<SettingsBrightness fontSize="small" />}
+                    style={{ padding: '1rem 2rem', width: '100%' }}
+                    value="system"
+                    onClick={handleThemeChange}
+                >
+                    System
+                </Button>
+                <Button
+                    startIcon={<DarkMode fontSize="small" />}
+                    style={{ padding: '1rem 2rem', borderRadius: '0px 12px 12px 0px', width: '100%' }}
+                    value="dark"
+                    onClick={handleThemeChange}
+                >
+                    Dark
+                </Button>
+            </ButtonGroup>
+        </Box>
+    );
+}
 
-    const handleTimeChange = (event: ChangeEvent<HTMLInputElement>) => {
-        useTimeFormatStore.getState().setTimeFormat(event.target.value == 'true');
+function TimeMenu() {
+    const handleTimeChange = (event: React.MouseEvent<HTMLButtonElement>) => {
+        useTimeFormatStore.getState().setTimeFormat(event.currentTarget.value == 'true');
     };
 
-    useEffect(() => {
-        AppStore.on('themeToggle', handleThemeChange);
+    return (
+        <Box sx={{ padding: '1rem 1rem 0 1rem', width: '100%' }}>
+            <Typography variant="h6" style={{ marginTop: '1.5rem', marginBottom: '1rem' }}>
+                Time
+            </Typography>
 
-        return () => {
-            AppStore.off('themeToggle', handleThemeChange);
-        };
-    }, []);
+            <ButtonGroup style={{ display: 'flex', placeContent: 'center', width: '100%' }}>
+                <Button
+                    style={{
+                        padding: '1rem 2rem',
+                        borderRadius: '12px 0px 0px 12px',
+                        width: '100%',
+                        fontSize: '12px',
+                    }}
+                    value="false"
+                    onClick={(value) => handleTimeChange(value)}
+                    fullWidth={true}
+                >
+                    12 Hour
+                </Button>
+                <Button
+                    style={{
+                        padding: '1rem 2rem',
+                        borderRadius: '0px 12px 12px 0px',
+                        width: '100%',
+                        fontSize: '12px',
+                    }}
+                    value="true"
+                    onClick={(value) => handleTimeChange(value)}
+                >
+                    24 Hour
+                </Button>
+            </ButtonGroup>
+        </Box>
+    );
+}
+
+function SettingsMenu() {
+    const [drawerOpen, setDrawerOpen] = useState(false);
 
     return (
         <>
             <Button
-                onClick={(event) => {
-                    setAnchorEl(event.currentTarget);
+                onClick={() => {
+                    setDrawerOpen(true);
```

Comment:
You could memo-ize this callback. But it's not a big deal I guess. Same difference